### DISCUSSION TITLE
Update `#[Reactive]` check to allow models to lazily load relations

### DISCUSSION THREAD

**Discussion by jplhomer** [livewire/livewire#8072] [3. Feature ideas]
Was chatting with @joshhanley in Discord about this:

1. You have a child Livewire component which accepts a Model as a mounted `#[Reactive]` property
2. You take care to not intentionally mutate any properties of the model
3. However, your components end up calling (directly or indirectly) code which lazily-loads a relation on that model, e.g. `$user->team->name`
4. This causes the `#[Reactive]` check to fail, since the objects no longer match.

While it makes sense to protect the integrity of the reactive property, is it possible we're being too strict when it comes to Eloquent models, in particular loaded relations? Livewire doesn't attempt to restore your relations for you, and none of those get persisted to the front-end anyway.

**Comment by thomas-alrek** [comment]
This. I was really struggling with this. I didn't see anything in the doumentation about loaded relationships breaking the diff check for Reactive props. I had to dig through the codebase to understand the issue. It would be perfect if this was somehow handled by Reactive directly. It should be possible to interrogate the model to see if an attribute is a relationship or not, and just ignore that from the diff.
